### PR TITLE
feat: add msc4028 to versions api

### PR DESCRIPTION
**Pull request by hanadi92**
Fixes/Closes: https://github.com/element-hq/synapse/issues/16786

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://element-hq.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
  - Feel free to credit yourself, by adding a sentence "Contributed by @github_username." or "Contributed by [Your Name]." to the end of the entry.
* [x] Pull request includes a [sign off](https://element-hq.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://element-hq.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://element-hq.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

Signed-off-by: Hanadi <hanadi.tamimi@gmail.com>

### CODE REVIEW

**Review by anoadragon453 — CHANGES_REQUESTED** [review]
Hi @hanadi92, thanks for your contribution, and the context behind it.

We will need to see the `/_matrix/clients/versions` endpoint and `org.matrix.msc4028` mentioned in MSC4028 first before we add it to Synapse, otherwise the implementation is not compliant with the MSC.

It appears as if @giomfo alluded to doing so in https://github.com/matrix-org/matrix-rust-sdk/pull/2840#issuecomment-1875587994, so I suggest following up there.

**Comment by giomfo** [comment]
@anoadragon453 @hanadi92 fyi I updated MSC4028 on this point: https://github.com/matrix-org/matrix-spec-proposals/pull/4028/commits/d3a1e87c26bd811372bf4819413ed685242612b7

**Comment by hanadi92** [comment]
Thanks @giomfo 
do I re-request a review? or are we waiting for something else?

File: changelog.d/16787.feature [hunk 1]
```diff
@@ -0,0 +1 @@
+Add msc4028 experimental feature config value to /_matrix/clients/versions api. Contributed by @hanadi92.
```
Comment (anoadragon453) [inline changelog.d/16787.feature:1]:
```suggestion
Advertise experimental support for [MSC4028](https://github.com/matrix-org/matrix-spec-proposals/pull/4028) through `/_matrix/clients/versions` if enabled. Contributed by @hanadi92.
```

**Review by anoadragon453 — APPROVED** [review]
Ah! Sorry, I had been waiting for https://github.com/matrix-org/matrix-spec-proposals/pull/4028#discussion_r1447392602 to be resolved, but had missed it when it happened. Thank you for the poke, and sorry for the delay.

This PR now looks good to me!

**Pull request merged**